README.md
# Flutter AdvancedRichText

## Intro

Flutter AdvancedRichText is the simple way to create RichText with variety of text having different textstyles. 

- [Features](#features)
- [Options](#options)
- [Usage](#usage)
- [Output](#output)



## Features

* You can create different text giving styles individually.
* You can create rich text in easiest way.
* You can use same style in the other text without writing any codes for that.

## Options

### For AdvancedRichText:

| Option | Description | Type | Required |
|---|---|---|:---:|
initialText | The text which you want to show at first | String | Yes
initialTextStyle | The textstyle which you want to show at first | TextStyle | No
secondaryTextStyle | The textstyle which you want to show in other text | TextStyle | No
textList | The different text which you want to show | TextSpanList | Yes

### For TextSpanList:

| Option | Description | Type | Required |
|---|---|---|:---:|
text | The text which you want to show along with | String | Yes
initialTextLike | True for having secondaryTextStyle same as initialTextStyle | Bool | No (default false)
textStyle | The textstyle which you want to show in other text | TextStyle | No
onTap | Function you want to be performed after click | Function | No

## Usage

```dart

AdvanceRichText(
            initialText: "Haven't liked this package, yet? ",
            secondaryTextStyle: TextStyle(
              color: Colors.green,
              fontWeight: FontWeight.w700,
            ),
            textList: [
              TextSpanList(
                text: "Please, do Like.",
              ),
            ],
          ),

```



```dart

AdvanceRichText(
          initialText: "I agree that ",
          secondaryTextStyle: TextStyle(
            color: Colors.green,
            fontWeight: FontWeight.w700,
          ),
          textList: [
            TextSpanList(
                text: "AdvancedRichText ",
                textStyle: TextStyle(
                    color: Colors.green[600],
                    fontWeight: FontWeight.w700,
                    fontSize: 24)),
            TextSpanList(text: "package ", initialTextLike: true),
            TextSpanList(
                text: "is the easiest way to create ",
                textStyle: TextStyle(
                    color: Colors.blue[500], fontWeight: FontWeight.w500)),
            TextSpanList(
                text: "different styled texts.",
                textStyle: TextStyle(
                    color: Colors.purple[200], fontWeight: FontWeight.w700)),
            TextSpanList(
                text: " Agreed?",
                textStyle:
                    TextStyle(color: Colors.red, fontWeight: FontWeight.w900)),
          ],
        ),

```
## Output


<img src="/example/image/image.png" width="500" height="500"/>



---

## Thanks

If something is missing or you want to add some feature, feel free to open a ticket or contribute!

[LICENSE: MIT](LICENSE)


CHANGELOG.md
## 0.0.1+1

* Docs updated


## 0.0.1 

* First release.


lib/advanced_rich_text.dart
// ignore_for_file: prefer_const_constructors

library advanced_rich_text;

import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';

class AdvanceRichText extends StatefulWidget {
  final String initialText;
  final TextStyle? initialTextStyle;
  final TextStyle? secondaryTextStyle;
  final List<TextSpanList> textList;

  const AdvanceRichText(
      {Key? key,
      required this.textList,
      required this.initialText,
      this.initialTextStyle,
      this.secondaryTextStyle})
      : super(
          key: key,
        );

  @override
  State<AdvanceRichText> createState() => _AdvanceRichTextState();
}

class _AdvanceRichTextState extends State<AdvanceRichText> {
  @override
  Widget build(BuildContext context) {
    return RichText(
        text: TextSpan(
            text: widget.initialText,
            style: widget.initialTextStyle ?? TextStyle(color: Colors.black),
            children: [
          ...widget.textList
              .map((e) => textSpanText(
                  textStyle: e.textStyle,
                  label: e.text,
                  first: e.initialTextLike,
                  onTapText: e.onTap))
              .toList()
        ]));
  }

  TextSpan textSpanText({
    required String label,
    bool first = false,
    TextStyle? textStyle,
    Function()? onTapText,
  }) {
    return TextSpan(
        text: label,
        style: first
            ? widget.initialTextStyle
            : (textStyle ?? widget.secondaryTextStyle),
        recognizer: TapGestureRecognizer()..onTap = onTapText ?? () {});
  }
}

class TextSpanList {
  String text;
  bool initialTextLike;
  TextStyle? textStyle;
  Function()? onTap;
  TextSpanList(
      {required this.text,
      this.initialTextLike = false,
      this.onTap,
      this.textStyle});
}


example/lib/main.dart
// ignore_for_file: prefer_const_constructors, prefer_const_literals_to_create_immutables

import 'package:flutter/material.dart';
import 'package:advanced_rich_text/advanced_rich_text.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      title: 'Flutter Demo',
      home: AdvanceRichTextPage(),
    );
  }
}

class AdvanceRichTextPage extends StatefulWidget {
  const AdvanceRichTextPage({Key? key}) : super(key: key);

  @override
  State<AdvanceRichTextPage> createState() => _AdvanceRichTextPageState();
}

class _AdvanceRichTextPageState extends State<AdvanceRichTextPage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: ListView(padding: const EdgeInsets.all(20), children: [
        AdvanceRichText(
          initialText: "I agree that ",
          secondaryTextStyle: TextStyle(
            color: Colors.green,
            fontWeight: FontWeight.w700,
          ),
          textList: [
            TextSpanList(
                text: "AdvancedRichText ",
                textStyle: TextStyle(
                    color: Colors.green[600],
                    fontWeight: FontWeight.w700,
                    fontSize: 24)),
            TextSpanList(text: "package ", initialTextLike: true),
            TextSpanList(
                text: "is the easiest way to create ",
                textStyle: TextStyle(
                    color: Colors.blue[500], fontWeight: FontWeight.w500)),
            TextSpanList(
                text: "different styled texts.",
                textStyle: TextStyle(
                    color: Colors.purple[200], fontWeight: FontWeight.w700)),
            TextSpanList(
                text: " Agreed?",
                textStyle:
                    TextStyle(color: Colors.red, fontWeight: FontWeight.w900)),
          ],
        ),
        SizedBox(
          height: 100,
        ),
        Expanded(
          child: AdvanceRichText(
            initialText: "Haven't liked this package, yet? ",
            secondaryTextStyle: TextStyle(
              color: Colors.green,
              fontWeight: FontWeight.w700,
            ),
            textList: [
              TextSpanList(
                text: "Please, do Like.",
              ),
            ],
          ),
        ),
      ]),
    );
  }
}
